File: beginner/february/TS/2025-02-25_ABC392_C.ts
import { readFileSync } from 'fs';
import { join } from 'path';

const input = readFileSync(join(__dirname, '..//Users/shunyonai/Documents/GitHub/competitive-programming/input.txt'), 'utf8')
    .trim()
    .split('\n');

// 1行目を数値として取得
const N = +input[0];

// 2行目をスペース区切りの数値配列として取得
const [a, b] = input[1].split(' ').map(Number);

// 3行目を文字列として取得
const S = input[2];

console.log(N);
console.log(a, b);
console.log(S);

File: problems/Archive/february/2025-02-25_ABC392_C.py
import io,sys
with open("/Users/shunyonai/Documents/GitHub/atcoder/input.txt") as TxtOpen:
    INPUT=TxtOpen.read()
sys.stdin=io.StringIO(INPUT)
# --------------------------------------------------------

# issue
# Qのi番目のインデックス検索
# インデックス番目のPの番号を検索
# 検索した番号-1番目のQの要素をリストに追加
# 上記をループ実行

# AIによるソースレビュー
# indexの使用により、O(N^2)の計算量
# 辞書テーブルを作成
# joinは最後に一度のみ使用する

N = int(input())
Q = list(map(int, input().split()))
P = list(map(int, input().split()))

P_index = {value: index for index, value in enumerate(P)}
ans_list = [str(P[Q[P_index[i]]-1]) for i in range(1, N+1)]

print(' '.join(ans_list))


# AIによる別解
# 辞書ではなく配列でインデックスを管理する
# メモリ効率が若干向上する
N = int(input())
Q = list(map(int, input().split()))
P = list(map(int, input().split()))

# Pの各値の位置を記録する配列
pos = [0] * (N + 1)
for i in range(N):
    pos[P[i]] = i

# 結果を求める
result = []
for i in range(1, N + 1):
    person_pos = pos[i]      # i番の人の位置
    looking_at = Q[person_pos]  # その人が見ている人の番号
    result.append(str(P[looking_at - 1]))

print(' '.join(result))